### PR TITLE
Tiling by EQUI7

### PR DESCRIPTION
**Pull request by SerRichard**
Currently the save results puts spatial extent into a single file, this consumes too much memory in the back-end per worker. This adds some default tiling so that computation is only done for extents which can be  held in memory. This is also a first step towards supporting other tile grids.

### CODE REVIEW

File: src/openeo_processes/utils.py [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
 import re
 from datetime import timezone, timedelta, datetime
 from typing import Any, Callable, Tuple
 
+import dask
 import numpy as np
 import xarray as xr
 import geopandas as gpd
+from equi7grid import equi7grid
+from osgeo import osr
```
Comment (LukeWeidenwalker) [inline src/openeo_processes/utils.py:12]: These two dependencies want to be recorded in setup.cfg!

Comment (LukeWeidenwalker) [inline src/openeo_processes/utils.py:12]: These two dependencies want to be recorded in setup.cfg!

File: src/openeo_processes/cubes.py [hunk 2]
```diff
@@ -211,105 +216,62 @@ def exec_xar(data, output_filepath='out', format='GTiff', options={}, write_prod
...
-            data.time.attrs.pop('units', None)
         if hasattr(data, 'grid_mapping'):
             data.attrs.pop('grid_mapping')
-        # end workaround
+
+        if 't' in data.dims:
+            # Group datasets by time. This will help with storing multiple files via dask.
+            times, datasets = zip(*data.groupby("t"))
+        else:
+            times, datasets = zip(*data.groupby("time"))
```
Comment (LukeWeidenwalker) [inline src/openeo_processes/cubes.py:248]: You've already renamed the time dimension to `t` above, don't think this if-statement if necessary!

**Comment by LukeWeidenwalker** [comment]
Bunch of comments - I've had a short read over the previous implementation and tried to figure out what was essential and what wasn't, but my current understanding of the odc setup isn't good enough to accurately judge whether this is missing anything crucial. @ValentinaHutter is probably in a better position to review these aspects, my approval on this one isn't that valuable!

File: src/openeo_processes/utils.py (hunk 1 above)
Comment (ValentinaHutter) [reply src/openeo_processes/utils.py:13]: I think it should be enough to add the two dependencies in the setup.cfg, GDAL should be installed in the Dockerfile before we install the requirements from openeo-processes-python. (See https://git.eodc.eu/eodc/eodc-api/api-airflow-celery/-/blob/master/build/celery-dask/Dockerfile)

File: src/openeo_processes/cubes.py [hunk 3]
```diff
@@ -211,105 +216,61 @@ def exec_xar(data, output_filepath='out', format='GTiff', options={}, write_prod
...
-            # TODO this must be improved once `rename_dimension` is supported!
-            if 'time' in data.coords:
-                for timestamp in data.time.values:
-                    data_at_timestamp = data.loc[dict(time=timestamp)]
-                    all_tmp.append(reformat_dataset(data_at_timestamp, timestamp))
-            else:
-                all_tmp.append(reformat_dataset(data))
+        formats = ['gtiff', 'netcdf', 'geotiff']
+        if format.lower() in formats:
+            format == format.lower()
```
Comment (LukeWeidenwalker) [inline src/openeo_processes/cubes.py:222]: This should be a single =, otherwise this just checks equality and doesn't do anything!

**Pull request merged**